### PR TITLE
Frontend: save project constraints and priorities

### LINKED ISSUE #218: Frontend: Save priorities and constraints
In the diagnostic phase, a user can select priorities (elements/pillars) and set constraints. These should be saved to a new "Project Area" object before sending to ForSys with Patchmax. The results of the ForSys run (namely project area polygons) will be saved to the same object.

This requires a screen in the frontend.

### PR DESCRIPTION
**Pull request by thequeenofspades**
## Changes
- Small change to `plan/views.py` in backend: accept float or int values for parameters, and cast to float before saving. This is necessary because Typescript doesn't have a float type, only a number type.
- When the user enters the scenario planning flow, frontend calls the `plan/create_project` API to create a new, empty project.
- When the user edits values in scenario planning (e.g. priorities), frontend calls the `plan/update_project` API to store the new parameters.
- Unit tests.
- Fixes #218 

## No UI change

### CODE REVIEW

**Comment by elsiel23** [comment]
LGTM for backend changes. Lmk if you'd like me to review the front end changes as well. 

Would having a PATCH interface for update_projects be helpful going forward?

**Comment by thequeenofspades** [comment]
Thanks Elsie! I'll let Cynthia take a look at the frontend changes.

Hmm, I think we can do without PATCH unless we find that there's too much being sent over the network. Right now the only big-ish thing is the project area geojson and that's added in the last step, so it probably won't be sent too many times.

File: src/interface/src/app/plan/create-scenarios/create-scenarios.component.spec.ts [hunk 1]
```diff
@@ -1,22 +1,45 @@
...
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { BehaviorSubject, of } from 'rxjs';
+import { PlanService } from 'src/app/services';
+import { Plan, Region } from 'src/app/types';
 
 import { PlanModule } from '../plan.module';
 import { CreateScenariosComponent } from './create-scenarios.component';
 
-describe('CreateScenariosComponent', () => {
+fdescribe('CreateScenariosComponent', () => {
```
Comment (lattemilk) [inline src/interface/src/app/plan/create-scenarios/create-scenarios.component.spec.ts:11]: remove fdescribe here :D

File: src/interface/src/app/plan/create-scenarios/create-scenarios.component.spec.ts [hunk 2]
```diff
@@ -40,6 +63,28 @@ describe('CreateScenariosComponent', () => {
...
+      max_slope: NaN,
+    });
+  });
+
+  it('should not update project if form is invalid', () => {
+    component.formGroups[1].markAsDirty();
+    component.formGroups[1].get('budgetForm.maxBudget')?.setValue(-1);
+
+    expect(fakePlanService.updateProject).toHaveBeenCalledTimes(0);
+
```
Comment (lattemilk) [inline src/interface/src/app/plan/create-scenarios/create-scenarios.component.spec.ts:85]: nit extra blank line?

File: src/interface/src/app/plan/create-scenarios/create-scenarios.component.ts [hunk 3]
```diff
@@ -93,10 +96,11 @@ export class CreateScenariosComponent {
   formGroups: FormGroup[];
   readonly PlanStep = PlanStep;
   panelExpanded: boolean = true;
+  projectId: number | undefined;
```
Comment (lattemilk) [inline src/interface/src/app/plan/create-scenarios/create-scenarios.component.ts:99]: This can be replaced with:
projectId?: number;

**Review by lattemilk — APPROVED** [review]

**Pull request merged**